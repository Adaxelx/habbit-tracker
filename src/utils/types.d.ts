export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: boolean;
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}
