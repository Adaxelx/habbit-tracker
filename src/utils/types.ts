export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
  handleDayChange: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: [
    {
      day: number;
      month: number;
      year: number;
      _id: string;
    },
  ];
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
  userId: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}

export type Color = { color?: string; id?: string };

export type NavProps = {
  header: string;
  moveDate: Function;
};

export interface DayCardProps {
  day: string;
  token: string | undefined;
}
