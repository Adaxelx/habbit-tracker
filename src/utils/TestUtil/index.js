export { default } from './TestUtil.ts';
