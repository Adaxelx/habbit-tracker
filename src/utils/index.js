export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
