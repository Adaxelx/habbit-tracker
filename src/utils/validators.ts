export const createRequiredObject = (name: string) => ({
  required: `Field ${name} is required.`,
});

export const validatePattern = (pattern: RegExp) => ({
  pattern: {
    value: pattern,
    message: 'Incorrect characters.',
  },
});

export const emailValidation = validatePattern(
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/,
);

export const createRestrictedLengthObject = (name: string) => ({
  ...createRequiredObject(name),
  minLength: {
    value: 6,
    message: `Field ${name} must have at least 6 characters.`,
  },
  maxLength: {
    value: 30,
    message: `Field ${name} must have maximum 30 characters.`,
  },
});

const timeValidFuntion = (timeLess: string | undefined, time: string | undefined) => {
  const message = 'Your time is not in boundaries.';
  if (timeLess && time) {
    const [hours, minutes] = time.split(':');
    const [hoursMin, minutesMin] = timeLess.split(':');
    if (hoursMin < hours) {
      return true;
    }
    if (hoursMin === hours) {
      if (minutesMin < minutes) {
        return true;
      }
      return message;
    }
    return message;
  }
  return message;
};

export const createTimeValidation = (min: string | undefined, max: string | undefined) => ({
  validate: {
    min: (value: string) => timeValidFuntion(min, value),
    max: (value: string) => timeValidFuntion(value, max),
  },
});
