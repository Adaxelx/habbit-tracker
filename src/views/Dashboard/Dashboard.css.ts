import styled from 'styled-components';
import { FlexColCenter } from 'styles';

export const StyledWrapper = styled(FlexColCenter)`
  height: 100vh;
  width: 100%;
  padding: 0 1rem;
  @media (min-width: ${({ theme }) => theme.media.tablet.s}) {
    flex-direction: row;
    justify-content: space-between;
    max-width: 1000px;
    margin: 0 auto;
  }
`;

export const StyledContentWrapper = styled(FlexColCenter)`
  @media (min-width: ${({ theme }) => theme.media.tablet.s}) {
    margin-right: ${({ theme }) => theme.margin.l};
  }
`;

export const StyledHeader = styled.h2`
  font-size: ${({ theme }) => theme.font.sizes.l};
  margin: ${({ theme }) => theme.margin.sm} 0;
  @media (max-width: ${({ theme }) => theme.media.phone.s}) {
    font-size: ${({ theme }) => theme.font.sizes.ml};
  }
`;

export const StyledDescription = styled.p`
  color: ${({ theme }) => theme.colors.text.secondary};
  text-align: justify;
  font-size: ${({ theme }) => theme.font.sizes.m};
  margin-bottom: ${({ theme }) => theme.margin.l};
  @media (max-width: ${({ theme }) => theme.media.phone.s}) {
    font-size: ${({ theme }) => theme.font.sizes.sm};
  }
`;

export const StyledSeparator = styled.p`
  font-size: ${({ theme }) => theme.font.sizes.sm};
  margin: ${({ theme }) => theme.margin.s};
`;
