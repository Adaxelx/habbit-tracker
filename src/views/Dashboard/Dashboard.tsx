import React from 'react';
import { Link } from 'react-router-dom';
import { Button, Container } from 'components';
import paths from 'constants/paths';
import {
  StyledHeader,
  StyledWrapper,
  FlexColCenter,
  StyledDescription,
  StyledSeparator,
  StyledContentWrapper,
} from './Dashboard.css';

const Dashboard = () => (
  <Container>
    <StyledWrapper>
      <StyledContentWrapper>
        <StyledHeader>Habbit tracker</StyledHeader>
        <StyledDescription>
          Track your progress in very simple and personalized app
        </StyledDescription>
      </StyledContentWrapper>
      <FlexColCenter>
        <Button as={Link} to={paths.REGISTER} size="m">
          Create account for free
        </Button>
        <StyledSeparator>or</StyledSeparator>
        <Button as={Link} to={paths.LOGIN} size="m">
          Log in
        </Button>
      </FlexColCenter>
    </StyledWrapper>
  </Container>
);

export default Dashboard;
