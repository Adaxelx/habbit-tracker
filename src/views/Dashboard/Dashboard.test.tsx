import React from 'react';
import { TestUtil } from 'utils';
import Dashboard from './Dashboard';

describe('Dashboard', () => {
  let util: TestUtil;
  beforeEach(() => {
    util = new TestUtil(<Dashboard />);
  });

  it('should match snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
});
