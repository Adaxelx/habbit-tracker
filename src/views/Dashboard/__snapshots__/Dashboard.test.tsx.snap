// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  color: #fff;
  background-color: #FFC136;
  border: 1px solid #707070;
  padding: 1.25rem;
  font-size: 1.5rem;
  min-width: 150px;
  display: inline-block;
  text-align: center;
  margin-top: 0;
  margin-left: 0;
  margin-right: 0;
}

.c6:hover {
  cursor: pointer;
}

.c6:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c0 {
  min-height: 100vh;
  width: 100%;
}

.c2 {
  height: 100vh;
  width: 100%;
  padding: 0 1rem;
}

.c4 {
  font-size: 3rem;
  margin: 1rem 0;
}

.c5 {
  color: #808080;
  text-align: justify;
  font-size: 1.5rem;
  margin-bottom: 1.75rem;
}

.c7 {
  font-size: 1.25rem;
  margin: 0.75rem;
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    max-width: 1000px;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.75rem;
  }
}

@media (max-width:320px) {
  .c4 {
    font-size: 2rem;
  }
}

@media (max-width:320px) {
  .c5 {
    font-size: 1.25rem;
  }
}

<main
    class="c0"
  >
    <section
      class="c1 c2"
    >
      <section
        class="c1 c3"
      >
        <h2
          class="c4"
        >
          Habbit tracker
        </h2>
        <p
          class="c5"
        >
          Track your progress in very simple and personalized app
        </p>
      </section>
      <section
        class="c1"
      >
        <a
          class="c6"
          href="/register"
        >
          Create account for free
        </a>
        <p
          class="c7"
        >
          or
        </p>
        <a
          class="c6"
          href="/login"
        >
          Log in
        </a>
      </section>
    </section>
  </main>
</DocumentFragment>
`;
